dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -0,0 +1,1078 @@
+            Field field;                                                                                            \
+            col_const->get(0, field);                                                                               \
+            String tmp = field.isNull() ? String("") : field.safeGet<String>();                                     \
+            if (col_const_data->isColumnNullable())                                                                 \
+            {                                                                                                       \
+                const auto * null_map = &(static_cast<const ColumnNullable &>(*(col_const_data)).getNullMapData()); \
+                Param<ParamString<true>, true>(param_name)(col_size, StringRef(tmp.data(), tmp.size()), null_map);  \
+                next_convertion;                                                                                       \
+            }                                                                                                       \
+            else                                                                                                    \
+            {                                                                                                       \
+                Param<ParamString<true>, false>(param_name)(col_size, col_const->getDataAt(0));                     \
+                next_convertion;                                                                                       \
+            }                                                                                                       \
+        }                                                                                                           \
+        else                                                                                                        \
+        {                                                                                                           \
+            CONVERT_NULL_STR_COL_TO_PARAM((param_name), (converted_col), next_convertion)                              \
+        }                                                                                                           \
+    } while (0);
+
+class FunctionStringRegexpBase
+{
+public:
+    static constexpr size_t REGEXP_MIN_PARAM_NUM = 2;
+
+    // Max parameter number the regexp_xxx function could receive
+    static constexpr size_t REGEXP_MAX_PARAM_NUM = 2;
+    static constexpr size_t REGEXP_LIKE_MAX_PARAM_NUM = 3;
+    static constexpr size_t REGEXP_INSTR_MAX_PARAM_NUM = 6;
+    static constexpr size_t REGEXP_REPLACE_MAX_PARAM_NUM = 6;
+    static constexpr size_t REGEXP_SUBSTR_MAX_PARAM_NUM = 5;
+
+    template <typename ExprT, typename MatchTypeT>
+    void memorize(const ExprT & pat_param, const MatchTypeT & match_type_param, TiDB::TiDBCollatorPtr collator) const
```

Comment:
Don't need to check null value here?

Reply:
> Don't need to check null value here?

const null case has been handled at the beginning of `executeImpl`, it's impossible to appear const null column

Reply:
> Don't need to check null value here?

resolved